controllers/cloud_config_sync_controller_test.go

```diff
@@ -0,0 +1,205 @@
+package controllers
+
+import (
+	"context"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	configv1 "github.com/openshift/api/config/v1"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/tools/record"
+	"k8s.io/utils/pointer"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+func getRequestForObject(object client.Object) reconcile.Request {
+	return reconcile.Request{
+		NamespacedName: client.ObjectKeyFromObject(object),
+	}
+}
+
+var _ = Describe("Cloud config sync controller", func() {
+	var r *CloudConfigReconciler
+	var rec *record.FakeRecorder
+
+	var infraCloudConfig *corev1.ConfigMap
+	var managedCloudConfig *corev1.ConfigMap
+	var syncedCloudConfigMap *corev1.ConfigMap
+
+	ctx := context.Background()
+
+	targetNamespaceName := testManagementNamespace
+
+	infraCloudConfName := "test-config"
+	infraCloudConfKey := "foo"
+
+	infrastructureResource := &configv1.Infrastructure{
+		Spec: configv1.InfrastructureSpec{
+			CloudConfig: configv1.ConfigMapFileReference{
+				Name: infraCloudConfName,
+				Key:  infraCloudConfKey,
+			},
+			PlatformSpec: configv1.PlatformSpec{
+				Type: "VSphere",
```

Comment:
In the configv1 package there are a number of different constants, would prefer to use one of those here instead if possible. Also, what is reading from the platform spec? We normally read from the platform status?